header/p2p/exchange.go

```diff
@@ -85,14 +85,15 @@ func (ex *Exchange) Head(ctx context.Context) (*header.ExtendedHeader, error) {
 	}
 
 	result := make([]*header.ExtendedHeader, 0, len(ex.trustedPeers))
+LOOP:
 	for range ex.trustedPeers {
 		select {
 		case h := <-headerCh:
 			if h != nil {
 				result = append(result, h)
 			}
 		case <-ctx.Done():
-			return nil, ctx.Err()
+			break LOOP
```

Comment:
We might also want to limit the Head call itself with a deadline. We don't do it where call `Head` in Syncer. The question is: should the user of the API do this himself, or should we do the limit for each Head call internally like we do it in the Server?

Reply:
Other than this Q, LGTM, clean solution

Reply:
I would suggest to let users decide themselves. We have done it for Server side as this is a dos vector - to request 1000 range request that will block the whole service. But I don't think it's relevant for the client side.

Reply:
Let's put the above into an issue bc it applies to every client-side call @Wondertan 

At the moment, let's leave it for user/caller to decide until we see it causes issues / can be problematic.

Reply:
@vgonkivs, pls make an issue

Reply:
https://github.com/celestiaorg/celestia-node/issues/1349